lib/errorResponseInterceptor.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * © Copyright IBM Corporation 2024. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function errorResponseInterceptor(axiosError) {
+  if (
+    axiosError?.response && // must have a response
+    axiosError?.response.status >= 400 &&
+    axiosError?.config.method !== 'head' &&
```

Comment:
This seems redundant to below check on data presence. Do we expect a case where we have "head" req with body?